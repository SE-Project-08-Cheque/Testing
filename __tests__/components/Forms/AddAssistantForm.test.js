// import dependencies
import React from "react";

// import react-testing methods
import { render, fireEvent, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom/extend-expect";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import AddAssistantForm from "../../../components/Forms/AddAssistantForm";

afterEach(cleanup);


it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <AddAssistantForm />,
    div
  );
});

it("adds assistant correctly", () => {
  const { getByTestId, getByText } = render(
    <AddAssistantForm />
  );

 
  fireEvent.change(getByTestId("firstname"),{ target: { value: 'mirissa' } })
  fireEvent.change(getByTestId("lastname"),{ target: { value: '23' } })
  fireEvent.change(getByTestId("email"),{ target: { value: '23' } })
  
  
  fireEvent.click(getByText("Add the Assistant")) 

  
});

it("matches snapshot", () => {
  const tree = renderer.create(
    <AddAssistantForm />
  );

  expect(tree).toMatchSnapshot();
});
