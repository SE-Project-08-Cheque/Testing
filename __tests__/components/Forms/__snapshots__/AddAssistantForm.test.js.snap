// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-1ceyi20"
>
  <form
    action="submit"
  >
    <div
      className="chakra-stack css-1n2jfo2"
    >
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <input
          aria-describedby="field-40-helptext"
          aria-label="Fisrt Name"
          aria-required={true}
          className="chakra-input css-0"
          data-testid="firstname"
          disabled={false}
          id="field-40"
          name="firstname"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="First Name"
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-40-helptext"
        >
          
        </div>
      </div>
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <input
          aria-describedby="field-41-helptext"
          aria-label="Last Name"
          aria-required={true}
          className="chakra-input css-0"
          data-testid="lastname"
          disabled={false}
          id="field-41"
          name="lastname"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Last Name"
          readOnly={false}
          required={true}
          type="text"
          value=""
        />
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-41-helptext"
        >
          
        </div>
      </div>
      <hr
        aria-orientation="horizontal"
        className="chakra-divider css-1upb9tn"
        role="separator"
      />
      <div
        className="chakra-form-control css-1kxonj9"
        role="group"
      >
        <input
          aria-describedby="field-42-helptext"
          aria-label="Email"
          aria-required={true}
          className="chakra-input css-0"
          data-testid="email"
          disabled={false}
          id="field-42"
          name="email"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Email"
          readOnly={false}
          required={true}
          type="email"
          value=""
        />
        <div
          className="chakra-form__helper-text css-1xnfdvb"
          id="field-42-helptext"
        >
          
        </div>
      </div>
      <button
        className="chakra-button css-oqfhsh"
        onClick={[Function]}
        type="submit"
      >
        Add the Assistant
      </button>
    </div>
  </form>
</div>
`;
